### DISCUSSION TITLE
Authelia - fatal error: Redis connection error

### DISCUSSION THREAD

**Discussion by jamesngob** [authelia/authelia#6374] [Q&A]
I have setup Nginx Proxy Manager and intend to use Authelia to provide 2-factor authentication for the exposed web services.
I copy the configuration.yml, modify it according to our environment and fire up. This is the error that I received and the process stops.

level=fatal msg="Redis connection error: dial tcp: lookup redis: i/o timeout" stack="github.com/authelia/authelia/v4/internal/commands/helpers.go:84 getProviders\ngithub.com/authelia/authelia/v4/internal/commands/root.go:72    cmdRootRun\ngithub.com/spf13/cobra@v1.6.1/command.go:920                    (*Command).execute\ngithub.com/spf13/cobra@v1.6.1/command.go:1044                   (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.6.1/command.go:968                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10         main\nruntime/proc.go:250                                             main\nruntime/asm_amd64.s:1594                                        goexit"

I have users_database.yml and all other required config files according to the Guide

Am I missing something here?
Thank you
James

**Comment by james-d-elliott** [comment]
Looks like there is a DNS lookup timeout (network layer) when performing a query for `redis`.

**Comment by Jerick00201** [comment]
I actually ran into this same error message when upgrading to the latest version. Moved back down to v4.37.4 and the issue is gone.

**Reply by james-d-elliott** [reply]
Very strange considering the diff has no changes to redis: https://github.com/authelia/authelia/compare/v4.37.4...v4.37.5

Can you try again?

**Reply by Jerick00201** [reply]
Switched back to the latest version and it's working fine now. Ran a docker pull and it looks like something was pulled from redis so perhaps that was the issue?

![image](https://github.com/authelia/authelia/assets/8799898/4d8d3f99-9caa-42c6-89cd-4adbcc76e5de)

**Reply by james-d-elliott** [reply]
It *could* be, if redis isn't in a healthy state then the DNS for it will not be available to Authelia (or anything else).
